components/ingest-service/migration/actions_migration.go

```diff
@@ -0,0 +1,143 @@
+package migration
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	chef "github.com/chef/automate/api/external/ingest/request"
+	"github.com/chef/automate/components/ingest-service/backend"
+)
+
+func (ms *Status) migrateAction() {
+	ms.total = 1
+	err := ms.SendAllActionsThroughPipeline()
+	if err != nil {
+		ms.updateErr(err.Error(), "Unable to re-insert actions")
+	}
+	ms.taskCompleted()
+
+	ms.finish("Migration Actions to current finished successfully")
+}
+
+func (ms *Status) SendAllActionsThroughPipeline() error {
```

Comment:
This was updated to re-ingest one of the "actions-*" indexes at a time and then delete it. This prevents all the actions from being re-ingested if the service dies or is killed.